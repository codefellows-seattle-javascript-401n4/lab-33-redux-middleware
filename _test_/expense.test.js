'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import Jest from 'jest';
import uuid from 'uuid/v1';
import ExpenseForm from '../src/components/expenseForm.js';

test('should pass with valid expense content', () => {
  let name = 'eggs';
  let price = 5;
  let createExpense = (state) => {
    expect(ExpenseForm.state().name).toBe(name);
    expect(ExpenseForm.state().price).toEqual(price);
  };
});

test('should fail with a change of the expense name', () => {
  const component = renderer.create(
    <ExpenseForm name="eggs"/>
  );
  const other = renderer.create(
    <ExpenseForm name="milk"/>
  );

  let tree = component.toJSON();
  let otherTree = other.toJSON();
  expect(tree).toMatchSnapshot();
  expect(otherTree).toEqual(tree);
});

test('should pass with an unique id', () => {
  const component = renderer.create(
    <ExpenseForm id={uuid()}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('should fail because non unique ids', () => {
  const component = renderer.create(
    <ExpenseForm id="1"/>
  );
  const other = renderer.create(
    <ExpenseForm id="1"/>
  );
  let tree = component.toJSON();
  let otherTree = otherTree.toJSON();
  expect(tree).toMatchSnapshot();
  expect(otherTree).toEqual(tree);
});
