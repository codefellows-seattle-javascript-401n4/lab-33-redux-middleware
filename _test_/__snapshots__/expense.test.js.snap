// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fail with a change of the expense name 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      id="expenseName"
    >
       Enter Your Expense Name 
    </label>
    <input
      htmlFor="expenseName"
      onChange={[Function]}
      value=""
    />
    <br />
    <label
      id="expensePrice"
    >
       Enter Your Expense Price 
    </label>
    $
    <input
      htmlFor="expensePrice"
      onChange={[Function]}
      type="number"
      value="$0"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;

exports[`should pass with an unique id 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      id="expenseName"
    >
       Enter Your Expense Name 
    </label>
    <input
      htmlFor="expenseName"
      onChange={[Function]}
      value=""
    />
    <br />
    <label
      id="expensePrice"
    >
       Enter Your Expense Price 
    </label>
    $
    <input
      htmlFor="expensePrice"
      onChange={[Function]}
      type="number"
      value="$0"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
